src/lib.rs

```diff
@@ -288,14 +303,23 @@ where
                         // longer need to cancel this activity, so we'll just ignore such orphaned
                         // cancellation.
                         if let Some(details) = self.outstanding_activity_tasks.get(&task_token) {
-                            return Ok(ActivityTask::from_ids(
-                                task_token.clone(),
+                            return Ok(ActivityTask::cancel_from_ids(
+                                task_token,
                                 details.activity_id.clone(),
                             ));
+                        } else {
+                            // TODO: Unless we return this JS seems to just give up on polling.
+                            //   -- we shouldn't need to return it?
```

Comment:
@bergundy I don't want to actually have to return this. This would be something good for you to look into why things get stuck when this isn't returned.